### PR TITLE
Add way to disable run and debug code lens provider

### LINKED ISSUE #464: Add way to disable run and debug buttons on main function definition.
I have a build where if you just attempt to run the main function, it fails to start, because I have some native dependencies that need to be configured. I am providing this build to a large amount of users, and it is confusing our users, as those buttons do not work, and they throw very confusing errors. Could a property be added to disable these buttons from showing up?

### PR DESCRIPTION
**Pull request by ThadHouse**
There are cases where it will never work properly, and it is confusing a chunk of my users that attempt to use it, and wonder why it fails.

Closes #464

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -364,6 +364,11 @@
           "type": "boolean",
           "description": "%java.debugger.configuration.enableHotCodeReplace.description%",
           "default": true
+        },
+        "java.debug.settings.enableRunDebugCodeLens": {
+          "type": "boolean",
+          "description": "Enable the run and debug code lens providers over main methods",
```
Comment (andxu) [inline package.json]: use %template% style of the most recent commit

Comment (ThadHouse) [reply package.json]: Fixed. I used google translate to get the italian translation

File: src/debugCodeLensProvider.ts [hunk 2]
```diff
@@ -11,11 +11,53 @@ import * as utility from "./utility";
...
+        this.debugCommand = vscode.commands.registerCommand(JAVA_DEBUG_COMMAND, debugJavaProgram);
+
+        const configuration = vscode.workspace.getConfiguration(JAVA_DEBUG_CONFIGURATION)
+        const isCodeLensEnabled = configuration.get<boolean>(ENABLE_CODE_LENS_VARIABLE);
+
+        if (isCodeLensEnabled) {
+            this.lensProvider = vscode.languages.registerCodeLensProvider(JAVA_LANGID, new DebugCodeLensProvider());
+        }
+
+        this.configurationEvent = vscode.workspace.onDidChangeConfiguration((event) =>  {
```
[the remark this answers is not in the document]
Comment (ThadHouse) [reply src/debugCodeLensProvider.ts]: Is that the standard in here? There are other places I saw looking around that did not type the parameter, so I wanted to match.

https://github.com/Microsoft/vscode-java-debug/blob/master/src/configurationProvider.ts#L21

[the remark this answers is not in the document]
Comment (testforstephen) [reply src/debugCodeLensProvider.ts]: It's better to do that. We need fix that place too.

[the remark this answers is not in the document]
Comment (ThadHouse) [reply src/debugCodeLensProvider.ts]: Ok. I pushed a fix for this one.

**Pull request merged**